crates/ruff_linter/src/rules/flake8_boolean_trap/helpers.rs

```diff
@@ -51,13 +51,29 @@ pub(super) fn is_allowed_func_def(name: &str) -> bool {
 /// Returns `true` if an argument is allowed to use a boolean trap. To return
 /// `true`, the function name must be explicitly allowed, and the argument must
 /// be either the first or second argument in the call.
-pub(super) fn allow_boolean_trap(func: &Expr) -> bool {
-    if let Expr::Attribute(ast::ExprAttribute { attr, .. }) = func {
-        return is_allowed_func_call(attr);
+pub(super) fn allow_boolean_trap(call: &ast::ExprCall) -> bool {
+    let func_name = match call.func.as_ref() {
+        Expr::Attribute(ast::ExprAttribute { attr, .. }) => attr.as_str(),
+        Expr::Name(ast::ExprName { id, .. }) => id.as_str(),
+        _ => return false,
+    };
+
+    // If the function name is explicitly allowed, then the boolean trap is
+    // allowed.
+    if is_allowed_func_call(func_name) {
+        return true;
     }
 
-    if let Expr::Name(ast::ExprName { id, .. }) = func {
-        return is_allowed_func_call(id);
+    // If the function appears to be a setter (e.g., `set_visible` or `setVisible`), then the
+    // boolean trap is allowed.
+    if call.arguments.args.len() == 1 {
+        if func_name
+            .strip_prefix("set_")
+            .and_then(|suffix| suffix.chars().next())
+            .is_some_and(|c| c == '_' || c.is_ascii_uppercase())
```

Comment:
I think this can be simplified to 
```suggestion
            .and_then(|suffix| suffix.starts_with(|c: char| c == '_' || c.is_ascii_uppercase()))
```